src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,68 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconSize } from "@components/Icon/Icon";
+import { ReactComponent as IconCaretDown } from "@components/Icon/Svg/CaretDown.svg";
+import { ReactComponent as IconCaretUp } from "@components/Icon/Svg/CaretUp.svg";
+import { Size } from "@utilities/enum";
+import { ReactElement, useState } from "react";
+import css from "./Dropdown.module.css";
+import DropdownMenu from "./DropdownMenu/DropdownMenu";
+import DropdownMenuItem, { MenuItem } from "./DropdownMenuItem/DropdownMenuItem";
+
+export interface DropdownProps {
+    menuItems: MenuItem[][] | MenuItem[];
+    onChange: (id?: string) => void;
+    activeItemId?: string;
+    placeholder?: string;
+    size?: Size.Small | Size.Large;
+}
+
+const getActiveItem = (menuItems: MenuItem[] | MenuItem[][], id: string) =>
+    menuItems.flat().find((item) => item.id === id) || null;
+
+export default function Dropdown({
+    menuItems,
+    onChange,
+    activeItemId = "",
+    placeholder = "",
+    size = Size.Small,
+}: DropdownProps): ReactElement<DropdownProps> {
+    const [open, setOpen] = useState(false);
+    const activeItem = getActiveItem(menuItems, activeItemId);
+
+    return (
+        <div className={css.wrapper}>
+            <div
```

Comment:
from a semantic perspective IMO this should be a `<button>` (you then potentially need to wrap your content within a `<span>` because grid/flex does not work as expected for a button 🙈 )